.github/workflows/main.yml

```diff
@@ -499,3 +499,127 @@ jobs:
         with:
           name: 'obs-studio-macos-${{ matrix.arch }}-notarized'
           path: '${{ github.workspace }}/${{ env.FILE_NAME }}'
+
+  macos_sparkle:
+    name: '04 - macOS Sparkle Updates'
+    runs-on: [macos-12]
+    needs: [macos_release]
+    if: ${{ startsWith(github.ref, 'refs/tags/') && github.event_name != 'pull_request' }}
+    strategy:
+      matrix:
+        arch: ['x86_64', 'arm64']
+    env:
+      HAVE_SPARKLE_KEY: ${{ secrets.SPARKLE_PRIVATE_KEY != '' }}
```

Comment:
I know it's not possible to check for a secret as a condition to run a job, does the same apply for a `needs` output? Because if that's possible we could check for the secret in one of the required steps and just skip the entire job.

Reply:
I think reading outputs *should* work in conditions based on https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability

Reply:
I now added this check to `macos_release` in a step that sets an output determining whether or not Sparkle should run. Tested this in my stripped-down test setup in my fork and validated that it works.